vars/sconsBuild.groovy

```diff
@@ -2,49 +2,201 @@
+    }
+    if (config['SRC_PREFIX']) {
+        scons_args += " SRC_PREFIX=${config['SRC_PREFIX']}"
+    } else if (config['target']) {
+        scons_args += " SRC_PREFIX=\${WORKSPACE}}"
+    }
+    if (config['REQUIRES']) {
+        scons_args += " REQUIRES=${config['REQUIRES']}"
+    } else if (config['target']) {
+        scons_args += " REQUIRES=${config['target']}"
+    }
+    //scons -c is not perfect so get out the big hammer
+    clean_files = "_build.external* install build"
+    if (config['clean']) {
+        clean_files = config['clean']
+    }
+    clean_cmd = "scons -c ${sconstruct}\n"
+    if (clean_files) {
+        clean_cmd += "rm -rf ${clean_files}\n"
+        if (config['target']) {
+            clean_cmd += "rm -rf ${config['target']}/${clean_files}\n"
+        }
+    }
+
+    prebuild = ''
+    if (config['prebuild']) {
+      prebuild = config['prebuild'] + '\n'
+    }
+    prefix_1 = ''
+
+    if (config['TARGET_PREFIX']) {
+      scons_args += " TARGET_PREFIX=${config['TARGET_PREFIX']}"
+      if (config['target_work']) {
+        target_dirs = ''
+        if (config['target_dirs']) {
+          target_dirs = config['target_dirs']
+        } else if (config['target']) {
+          target_dirs = config['target']
+        }
+        target_work = config['target_work']
+        prefix_1 =
+          """for new_dir in ${target_dirs}; do
+               mkdir -p "\${WORKSPACE}/${target_work}/\${new_dir}"
+               rm -f "${config['TARGET_PREFIX']}/\${new_dir}"
+               ln -s "\${WORKSPACE}/${target_work}/\${new_dir}" \
+                 "${config['TARGET_PREFIX']}/\${new_dir}"
+             done
+             """
+      }
+    }
+    if (config['scons_args']) {
+        scons_args += ' ' + config['scons_args']
+    }
 
     script = '''if git show -s --format=%B | grep "^Skip-build: true"; then
                     exit 0
                 fi
-                scons -c
-                # scons -c is not perfect so get out the big hammer
-                rm -rf _build.external install build
-                SCONS_ARGS="--update-prereq=all --build-deps=yes USE_INSTALLED=all install"
-                # the config cache is unreliable so always force a reconfig
-                # with "--config=force"
-                if ! scons --config=force $SCONS_ARGS; then
-                    rc=\${PIPESTATUS[0]}
-                    echo "$SCONS_ARGS failed"
-                    cat config.log || true
-                    exit \$rc
-                fi'''
-    rc = sh(script: script, returnStatus: true)
-
+                '''
+    script += clean_cmd
+    script += 'SCONS_ARGS="' + scons_args +'"\n'
+    script += '''# the config cache is unreliable so always force a reconfig
+                 # with "--config=force"
+                 if ! scons --config=force $SCONS_ARGS; then
+                     rc=\${PIPESTATUS[0]}
+                     echo "$SCONS_ARGS failed"
+                     cat config.log || true
```

Comment:
This doesn't work for projects that use an _arch_.  I have a patch to pipeline-lib for _arch_-using projects and a patch for daos to make it an _arch_-using project but it seems the concenus is actually in the other direction, to remove the _arch_ support from iof and cart.  I'm just not there yet.

Reply:
I do not recall making a change that would affect projects using an arch except for adding a wildcard for the cleanup.

Reply:
Yeah.  Probably alright to just leave it as is for now since it will correct for all projects once _arch_ support is removed from iof and cart.